scheduler/src/cook/components.clj

```diff
@@ -84,35 +84,38 @@
 
 (def mesos-scheduler
   {:mesos-scheduler (fnk [[:settings mesos-master mesos-master-hosts mesos-leader-path mesos-failover-timeout mesos-principal mesos-role mesos-framework-name offer-incubate-time-ms mea-culpa-failure-limit fenzo-max-jobs-considered fenzo-scaleback fenzo-floor-iterations-before-warn fenzo-floor-iterations-before-reset task-constraints riemann mesos-gpu-enabled rebalancer good-enough-fitness] mesos-datomic mesos-datomic-mult curator-framework mesos-pending-jobs-atom]
-                         (try
-                           (Class/forName "org.apache.mesos.Scheduler")
-                           ((lazy-load-var 'cook.mesos/start-mesos-scheduler)
-                            mesos-master
-                            mesos-master-hosts
-                            curator-framework
-                            mesos-datomic
-                            mesos-datomic-mult
-                            mesos-leader-path
-                            mesos-failover-timeout
-                            mesos-principal
-                            mesos-role
-                            mesos-framework-name
-                            offer-incubate-time-ms
-                            mea-culpa-failure-limit
-                            task-constraints
-                            (:host riemann)
-                            (:port riemann)
-                            mesos-pending-jobs-atom
-                            mesos-gpu-enabled
-                            rebalancer
-                            {:fenzo-max-jobs-considered fenzo-max-jobs-considered
-                             :fenzo-scaleback fenzo-scaleback
-                             :fenzo-floor-iterations-before-warn fenzo-floor-iterations-before-warn
-                             :fenzo-floor-iterations-before-reset fenzo-floor-iterations-before-reset
-                             :good-enough-fitness good-enough-fitness})
-                           (catch ClassNotFoundException e
-                             (log/warn e "Not loading mesos support...")
-                             nil)))})
+                         (let [make-mesos-driver-fn (partial (lazy-load-var 'cook.mesos/make-mesos-driver)
```

Comment:
The `lazy-load-var` are a way to avoid requiring namespaces initially which has two consequences (I explicitly didn't use "benefits" here):
1. AOT compilation can pass even if the lazy loaded namespace can't be compiled (since it is done at run time)
2. It makes initial start up time faster since not everything is loaded immediately

I'm not convinced either of these are worth the headache but I don't think this CR is the place to address it.